File: performance.kt
fun main() {
    val outerIterations = 10000
    val innerIterations = 100000
    var result: Long = 0

    println("Iniciando 1,000 millones de iteraciones con bucles anidados...")

    val startTime = System.nanoTime()

    for (i in 0 until outerIterations) {
        for (j in 0 until innerIterations) {
            result++
        }
    }

    val endTime = System.nanoTime()
    val executionTimeSeconds = (endTime - startTime) / 1_000_000_000.0

    println("\nResultado final: $result")
    println("Tiempo de ejecución: %.2f segundos".format(executionTimeSeconds))
}

File: performance.swift
import Foundation

let outerIterations = 10000
let innerIterations = 100000
var result = 0

print("Iniciando 1,000 millones de iteraciones con bucles anidados...")

let startTime = Date() // Inicio del timer

for _ in 0..<outerIterations {
    for _ in 0..<innerIterations {
        result += 1
    }
}

let endTime = Date() // Fin del timer
let executionTime = endTime.timeIntervalSince(startTime) // Diferencia en segundos

print("\nResultado final: \(result)")
print("Tiempo de ejecución: \(executionTime) segundos")
